### ISSUE TITLE
Evaluate timeout functionality

### ISSUE THREAD

**Issue by SwiftedMind** [SwiftedMind/Processed#1]
It would be nice to have the option to set a timeout for a process or loadable.

## Steps

- [x] Investigate
- [x] Build prototype
- [x] Implement
- [x] Add examples
- [x] Add Unit Tests
- [x] Update Readme

**Comment by SwiftedMind** [comment]
API could look something like this, which could even be expanded to a more general use case:

```swift
 $items.load(timeoutChecks: [.seconds(1), .seconds(5)]) {
   try await fetchItems()
 } timeoutCheck: { duration in
   if duration == .seconds(1) {
     showLoadingDelayedState = true
   } else {
     $items.cancel()
     showLoadingTimeoutState = true
   }
 }
```

**Comment by SwiftedMind** [comment]
Implemented with 2.0.0
